x-pack/plugins/security/server/lib/authorization/privileges/privileges.ts

```diff
@@ -86,3 +87,34 @@ export function privilegesFactory(actions: Actions, xpackMainPlugin: XPackMainPl
     },
   };
 }
+
+function applyAutomaticPrivilegeGrants(feature: Feature): Feature {
```

Comment:
This is the cleanest solution I've come up with so far. I originally had this taking place in `xpack_main` as part of `registerFeature()`, but it didn't feel right to have `xpack_main` concerned with security. The benefit of having it there was that it felt more noticeable to me, and less like "magic".

My first approach was to have this logic within `actions.savedObject`, but that ended up getting messy pretty quickly. For one, it required that I modify the `FeaturePrivilegeBuilder.getActions` signature to include the privilege ID, so that it knew when to include these additional saved object actions. That signature is used by all generators, and it felt unnecessary and messy to include that everywhere for this one use case.
It also buried this logic deep in the authz internals, and I didn't want to make maintainers go hunting for this.

I'm open to suggestions here

Reply:
I think your assessment of where to put this is fair, it doesn't really fit cleanly anywhere. Let's leave it where it is for now and see how things develop, I'm hoping there's a cleaner place to put this long-term.